src/asar/libsmw.cpp

```diff
@@ -403,19 +404,33 @@ bool openrom(const char * filename, bool confirm)
 	return true;
 }
 
-void closerom(bool save)
+unsigned int closerom(bool save)
 {
+	unsigned int romCrc = 0;
 	if (thisfile && save && romlen)
 	{
 		fseek(thisfile, header*512, SEEK_SET);
 		fwrite(const_cast<unsigned char*>(romdata), 1, (size_t)romlen, thisfile);
+
+		// do a quick re-read of the header, and include that in the crc32 calculation if necessary
+		{
+			unsigned char* filedata = (unsigned char*)malloc(sizeof(unsigned char) * (romlen + header * 512));
+			if (header)
+			{
+				fseek(thisfile, 0, SEEK_SET);
+				fread(filedata, sizeof(unsigned char), 512, thisfile);
+			}
+			memcpy(filedata + (header * 512), romdata, (size_t)romlen);
+			romCrc = crc32(filedata, (unsigned int)(romlen + header * 512));
+			delete filedata;
```

Comment:
Things allocated with `malloc` should be freed with `free`. You should only use `delete` for things allocated with `new`.
```suggestion
			free(filedata);
```